src/System.Text.Json/src/System/Text/Json/Writer/Utf8JsonWriter.WriteValues.Comment.cs

```diff
@@ -8,60 +8,49 @@
 
 namespace System.Text.Json
 {
-    public ref partial struct Utf8JsonWriter
+    public sealed partial class Utf8JsonWriter
     {
+        private static char[] s_singleLineCommentDelimiter = new char[2] { '*', '/' };
+        private static ReadOnlySpan<byte> SingleLineCommentDelimiterUtf8 => new byte[2] { (byte)'*', (byte)'/' };
+
         /// <summary>
         /// Writes the string text value (as a JSON comment).
         /// </summary>
         /// <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within /*..*/.</param>
-        /// <param name="escape">If this is set to false, the writer assumes the value is properly escaped and skips the escaping step.</param>
+        /// <remarks>
+        /// The comment value is not escaped before writing.
+        /// </remarks>
         /// <exception cref="ArgumentException">
-        /// Thrown when the specified value is too large.
+        /// Thrown when the specified value is too large OR if the given string text value contains a comment delimiter (i.e. */).
         /// </exception>
-        public void WriteCommentValue(string value, bool escape = true)
-            => WriteCommentValue(value.AsSpan(), escape);
+        public void WriteCommentValue(string value)
+            => WriteCommentValue(value.AsSpan());
 
         /// <summary>
         /// Writes the UTF-16 text value (as a JSON comment).
         /// </summary>
         /// <param name="value">The UTF-16 encoded value to be written as a UTF-8 transcoded JSON comment within /*..*/.</param>
-        /// <param name="escape">If this is set to false, the writer assumes the value is properly escaped and skips the escaping step.</param>
+        /// <remarks>
+        /// The comment value is not escaped before writing.
+        /// </remarks>
         /// <exception cref="ArgumentException">
-        /// Thrown when the specified value is too large.
+        /// Thrown when the specified value is too large OR if the given UTF-16 text value contains a comment delimiter (i.e. */).
         /// </exception>
-        public void WriteCommentValue(ReadOnlySpan<char> value, bool escape = true)
+        public void WriteCommentValue(ReadOnlySpan<char> value)
         {
             JsonWriterHelper.ValidateValue(value);
 
-            if (escape)
+            if (value.IndexOf(s_singleLineCommentDelimiter) != -1)
             {
-                WriteCommentEscape(value);
+                ThrowHelper.ThrowArgumentException_InvalidCommentValue();
```

Comment:
We still disallow embedded end of comment delimiter that might let the user escape out of the comment.

Reply:
FYI @GrabYourPitchforks